node/src/server-modules/GlideJson.ts

```diff
@@ -0,0 +1,172 @@
+/**
+ * Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0
+ */
+
+import { DecoderOption, GlideString } from "../BaseClient";
+import { ConditionalChange } from "../Commands";
+import { GlideClient } from "../GlideClient";
+import { GlideClusterClient } from "../GlideClusterClient";
+
+export type ReturnTypeJson = GlideString | GlideString[];
+/**
+ * Represents options for formatting JSON data, to be used in the [JSON.GET](https://valkey.io/commands/json.get/) command.
+ */
+export interface JsonGetOptions {
+    /** The path or list of paths within the JSON document. Default is root `$`. */
+    paths?: GlideString[];
+    /** Sets an indentation string for nested levels. Defaults to None. */
+    indent?: GlideString;
+    /** Sets a string that's printed at the end of each line. Defaults to None. */
+    newline?: GlideString;
+    /** Sets a string that's put between a key and a value. Defaults to None. */
+    space?: GlideString;
+}
+
+function _jsonGetOptionsToArgs(options: JsonGetOptions): GlideString[] {
+    const result: GlideString[] = [];
+
+    if (options.paths !== undefined) {
+        result.push(...options.paths);
+    }
+
+    if (options.indent !== undefined) {
+        result.push("INDENT", options.indent);
+    }
+
+    if (options.newline !== undefined) {
+        result.push("NEWLINE", options.newline);
+    }
+
+    if (options.space !== undefined) {
+        result.push("SPACE", options.space);
+    }
+
+    return result;
+}
+
+export class GlideJson {
+    /**
+     * Sets the JSON value at the specified `path` stored at `key`.
+
+        * See https://valkey.io/commands/json.set/ for more details.
+        *
+        * @param key - The key of the JSON document.
+        * @param path - Represents the path within the JSON document where the value will be set.
+        *      The key will be modified only if `value` is added as the last child in the specified `path`, or if the specified `path` acts as the parent of a new child being added.
+        * @param value - The value to set at the specific path, in JSON formatted bytes or str.
+        * @param conditionalChange - Set the value only if the given condition is met (within the key or path).
+        *      Equivalent to [`XX` | `NX`] in the Redis API. Defaults to null.
+        * 
+        * @returns If the value is successfully set, returns OK.
+        *       If `value` isn't set because of `conditionalChange`, returns null.
```

Comment:
```suggestion
        * @returns If the value is successfully set, returns `"OK"`.
        *       If `value` isn't set because of `conditionalChange`, returns `null`.
```